src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseGradingService.java

```diff
@@ -162,7 +163,9 @@ public Result processNewProgrammingExerciseResult(@NotNull ProgrammingExercisePa
         try {
             ContinuousIntegrationResultService ciResultService = continuousIntegrationResultService.orElseThrow();
             var buildResult = ciResultService.convertBuildResult(requestBody);
+
             checkCorrectBranchElseThrow(participation, buildResult);
+            checkHasCommitHashElseThrow(buildResult);
```

Comment:
I feel like this shouldn't be part of "processing", but it happened before; what do you think?

Reply:
This method is used twice (once in the endpoint and once by LocalCI) and I wanted to always include this check, so here is basically the earliest possible point.

Reply:
Rechecking the current implementation. It doesn't fit any different way.